packages/push/src/PushRegistration.ts

```diff
@@ -24,249 +22,38 @@ export interface PushRegistrationOptions {
  * let registration = new PushRegistration();
  * registration.register("myAppleOrFirebaseToken");
  */
-export class PushRegistration {
-
-  public static readonly TYPE: string = "push";
-  public static readonly REGISTRATION_DATA_KEY = "UPS_REGISTRATION_DATA";
+export class PushRegistration implements PushRegistrationInterface {
 
+  /**
+   * Calls to this method are ignored if webpush are used.
+   * @param onMessageReceivedCallback
+   */
   public static onMessageReceived(onMessageReceivedCallback: OnMessageReceivedCallback) {
-    PushRegistration.onMessageReceivedCallback = onMessageReceivedCallback;
+    PushRegistrationCordovaImpl.onMessageReceived(onMessageReceivedCallback);
```

Comment:
We should delegate it. Also It's not going to work for WebPush

Reply:
@danielpassos It delegates it to the `PushRegistrationCordovaImpl`. In the comment it is documented that calls to this method are ignored for WebPush.

The whole goal of using delegates here is to make the outer interface unchanged and, unfortunately, the old interface had this static method, so I had to leave it there to keep the interface unchanged.

Reply:
I added a `console.warn` message that is visualised if calling that method while using webpush

Reply:
No need to have that comment anymore: with the latest changes it is now used for webpush too.